File: app/src/main/java/algorithm/study/dohyun/boj/Boj_10773.java
package algorithm.study.dohyun.boj;
import java.util.*;
import java.io.*;
import java.util.Stack;
public class Boj_10773 {
    public static void main(String[] args) throws IOException {

        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st ;

        int K = Integer.parseInt(br.readLine());
        Stack<Integer> stk = new Stack<>();

        for(int i=0 ; i< K; i++){
            st = new StringTokenizer(br.readLine());
            int num = Integer.parseInt(st.nextToken());

            if( num == 0 ) {
                stk.pop();
                continue;
            }
            stk.push(num);
        }
        int answer = 0 ;
        for(Integer i: stk)
            answer+= i;
        System.out.println(answer);
    }
}

File: app/src/main/java/algorithm/study/devjk_sample/baekjoon/Boj_10773.kt
package algorithm.study.devjk_sample.baekjoon

import java.util.*

/**
 * 백준 - 제로
 *
 * @link https://www.acmicpc.net/problem/10773
 */
class Boj_10773 {

    fun solve(): Int {
        val k = readln().toInt()
        var st = Stack<Int>()
        for (i in 1..k) {
            val num = readln().toInt()
            if (num == 0) {
                st.pop()
                continue
            }
            st.push(num)
        }

        var ans = 0
        while (!st.empty()) {
            ans += st.pop()
        }
        return ans
    }
}

fun main(args: Array<String>) {
    val sol = Boj_10773()
    print(sol.solve())
}